File: dotnet/Example/Sort/Sort.cs
namespace Example.Sort
{
    public class Sort
    {
        private static int TEST = 111;
        static void Main(string[] args)
        {
            Otterop.Lang.String quick = Otterop.Lang.String.Wrap("quick");
            Otterop.Lang.String heap = Otterop.Lang.String.Wrap("heap");
            Otterop.Lang.Array<Otterop.Lang.String> argsArray = Otterop.Lang.Array.WrapString(args);
            if (argsArray.Size() < 1)
            {
                Otterop.Io.Console.Println(Otterop.Lang.String.Wrap("Usage: <quick|heap> ...values"));
                return ;
            }

            Otterop.Lang.String sortAlgorithm = argsArray.Get(0);
            bool isQuick = sortAlgorithm.CompareTo(quick) == 0;
            bool isHeap = sortAlgorithm.CompareTo(heap) == 0;
            if (!(isQuick || isHeap))
            {
                Otterop.Io.Console.Println(Otterop.Lang.String.Wrap("Usage: <quick|heap> ...values"));
                return ;
            }

            argsArray = argsArray.Slice(1, argsArray.Size());
            Example.Sort.Service.SortService sortService = new Example.Sort.Service.QuicksortService();
            if (isHeap)
            {
                sortService = new Example.Sort.Service.HeapsortService();
            }

            sortService.Sort(argsArray, 0, argsArray.Size());
            for (int i = 0; i < argsArray.Size(); i++)
            {
                Otterop.Io.Console.Println(argsArray.Get(i));
            }

            Otterop.Lang.Result.Of(Otterop.Lang.String.Wrap(""), Otterop.Lang.String.Wrap(""));
        }

    }

}

File: python/example/sort/sort.py
import sys
from example.sort.service.heapsort_service import HeapsortService
from example.sort.service.quicksort_service import QuicksortService
from example.sort.service.sort_service import SortService
from otterop.lang.array import Array
from otterop.lang.string import String
from otterop.lang.result import Result
from otterop.io.console import Console
println = Console.println

class Sort:
    test = 111

    @staticmethod
    def main(args):
        quick = String.wrap("quick")
        heap = String.wrap("heap")
        args_array = Array.wrap_string(args)
        if args_array.size() < 1:
            println(String.wrap("Usage: <quick|heap> ...values"))
            return 
        sort_algorithm = args_array.get(0)
        is_quick = sort_algorithm.compare_to(quick) == 0
        is_heap = sort_algorithm.compare_to(heap) == 0
        if not (is_quick or is_heap):
            println(String.wrap("Usage: <quick|heap> ...values"))
            return 
        args_array = args_array.slice(1, args_array.size())
        sort_service = QuicksortService()
        if is_heap:
            sort_service = HeapsortService()
        sort_service.sort(args_array, 0, args_array.size())
        i = 0
        while i < args_array.size():
            println(args_array.get(i))
            i += 1
        Result.of(String.wrap(""), String.wrap(""))

if __name__ == "__main__":
    Sort.main(sys.argv[1:])
